conformance/utils/kubernetes/helpers.go

```diff
@@ -297,18 +300,24 @@ func GatewayAndHTTPRoutesMustBeAccepted(t *testing.T, c client.Client, timeoutCo
 					SectionName: listener,
 				},
 				ControllerName: v1beta1.GatewayController(controllerName),
-				Conditions: []metav1.Condition{
-					{
-						Type:   string(v1beta1.RouteConditionAccepted),
-						Status: metav1.ConditionTrue,
-						Reason: string(v1beta1.RouteReasonAccepted),
-					},
-				},
+				Conditions: []metav1.Condition{{
+					Type:   string(v1beta1.RouteConditionAccepted),
+					Status: metav1.ConditionTrue,
+					Reason: string(v1beta1.RouteReasonAccepted),
+				}},
 			})
 		}
 		HTTPRouteMustHaveParents(t, c, timeoutConfig, routeNN, parents, namespaceRequired)
 	}
 
+	resolvedRefsCondition := metav1.Condition{
+		Type:   string(v1beta1.ListenerConditionResolvedRefs),
+		Status: metav1.ConditionTrue,
+		Reason: "", // any reason
+	}
+
+	GatewayListenersMustHaveCondition(t, c, timeoutConfig, gw.NamespacedName, resolvedRefsCondition)
```

Comment:
Since we are already adding the `ResolvedRefs` condition here, I think we should check all the positive-polarity conditions:
- `ResolvedRefs`
- `Accepted`
- `Programmed`

Ideally, `GatewayListenersMustHaveCondition` should check the three above conditions to be set and true, before going on. I've created https://github.com/kubernetes-sigs/gateway-api/issues/2415 for this purpose, as it looks like a lack of consistency in our suite. We can address it in another PR, as it is a bit out of scope for this PR, but I wanted to highlight this aspect here as well.